apps/frontend/src/app/components/level-indicator/level-indicator.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { XpSystemsService } from '../../services/xp-systems.service';
+
+@Component({
+  selector: 'm-level-indicator',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './level-indicator.component.html'
+})
+export class LevelIndicatorComponent implements OnInit {
+  protected level: number;
+  protected levelColor: string;
+  protected prestige: number;
+  protected prestigeIcon: string;
+  protected textColor: string;
+  protected imageClass: string;
+
+  @Input()
+  totalLevel: number;
+
+  constructor(private readonly xpService: XpSystemsService) {}
+
+  ngOnInit() {
+    this.level = this.xpService.getInnerLevel(this.totalLevel);
+    this.prestige = this.xpService.getPrestige(this.totalLevel);
+    this.prestigeIcon = this.xpService.getPrestigeIcon(this.prestige);
+
+    this.levelColor = this.getLevelColor(this.level);
+
+    this.textColor =
+      this.levelColor === '#ffffff' ? 'text-black' : 'text-white';
+    this.imageClass = this.levelColor === '#ffffff' ? 'invert-0' : 'invert';
+  }
+
+  private getLevelColor(level: number): string {
+    const levelColors = [
+      { max: 50, color: '#5d5d5d' },
+      { max: 100, color: '#e84855' },
+      { max: 150, color: '#ff8645' },
+      { max: 200, color: '#ffd000' },
+      { max: 250, color: '#47c27c' },
+      { max: 300, color: '#3f4aca' },
+      { max: 350, color: '#873de1' },
+      { max: 400, color: '#d34a8d' },
+      { max: 450, color: '#3185fc' },
+      { max: 500, color: '#161616' },
+      { max: Infinity, color: '#ffffff' }
+    ];
```

Comment:
Not that we really care about perf here, but I'd still move this to a `private static readonly` field of this class so you're not creating this same array every time this method is called.